drivers/at24cxxx/at24cxxx.c

```diff
@@ -204,7 +204,7 @@ size_t at24cxxx_write(const at24cxxx_t *dev, uint32_t pos, const void *data,
     size_t wr = 0;
     while (wr < len) {
         i2c_acquire(dev->params.i2c);
-        int w = _write(dev, pos + wr, data + wr, len - wr);
+        int w = _write(dev, pos + wr, (const uint8_t *)data + wr, len - wr);
```

Comment:
You don't need to cast a `void*`.

Reply:
cppcheck stated "undefined behaviour" there, because the void pointer is being incremented. It may not be absolutely necessary for proper functioning, but I think one should not do pointer arithmetic with void pointers.

Reply:
Oh good catch then!
Incrementing the `void*` left me pondering a bit too, I first thought it would be incremented by `sizeof(void*)`, but it turns out gcc always incremented it by 1.
Convenient, I thought, but if it's actually undefined the cast is of course necessary.